packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -488,9 +489,12 @@ export const DatePickerMixin = (subclass) =>
         this._selectDate(e.detail.date);
       });
 
-      // Keep focus attribute in focusElement for styling
+      // Set focus-ring attribute when moving focus to the overlay
+      // by pressing Tab or arrow key, after opening it on click.
       this._overlayContent.addEventListener('focusin', () => {
-        this._setFocused(true);
+        if (this._keyboardActive) {
```

Comment:
This check is needed to avoid `focus-ring` from getting removed on date click. Currently, it happens because `FocusMixin` is also used in `vaadin-month-calendar` which resets `keyboardActive` flag.